src/vaadin-grid-drag-and-drop-mixin.html

```diff
@@ -0,0 +1,257 @@
+              this._dropLocation = dropAbove ? DropLocation.ABOVE : DropLocation.BELOW;
+
+            } else if (this.dropMode === DropMode.ON_TOP_OR_BETWEEN) {
+              if (e.clientY - rowRect.top < rowRect.height / 3) {
+                this._dropLocation = DropLocation.ABOVE;
+              } else if (e.clientY - rowRect.top > (rowRect.height / 3) * 2) {
+                this._dropLocation = DropLocation.BELOW;
+              }
+            }
+          }
+
+          if (this._dropLocation === DropLocation.EMPTY) {
+            this.setAttribute('row-drag', '');
+          } else if (row) {
+            this._dragOverItem = row._item;
+            row.setAttribute('row-drag', this._dropLocation);
+          }
+        }
+      }
+
+      _clearDragStyles() {
+        this.removeAttribute('row-drag');
+        Array.from(this.$.items.children).forEach(row => row.removeAttribute('row-drag'));
+      }
+
+      _onDrop(e) {
+        e.stopPropagation();
+        const data = e.dataTransfer.getData('text');
+        this._clearDragStyles();
+
+        const event = new CustomEvent('grid-drop', {
+          bubbles: e.bubbles,
+          cancelable: e.cancelable,
+          detail: {
+            item: this._dragOverItem,
+            dropLocation: this._dropLocation,
+            data
+          }
+        });
+        this.dispatchEvent(event);
+      }
+
+      formatDragData(items) {
+        // TODO: Change this to either get the the default data from
+        // the physical cell content OR document properly that column "path"
+        // is needed to genrate the default payload
+        return items.map(item => {
+          return this._columnTree.slice(0).pop()
+            .sort((a, b) => {
+              return a._order > b._order ? 1 : -1;
+            })
+            .filter(col => col.path)
+            .map(col => this.get(col.path, item))
+            .join('\t');
+        }).join('\n');
+      }
+
+      _updateItem(row, item) {
+        super._updateItem && super._updateItem(row, item);
+
+        const draggableElements = window.ShadyDOM
+          ? [row]
+          : Array.from(row.children).map(cell => cell._content);
+
+        draggableElements.forEach(e => {
+          if (this.rowsDraggable) {
+            e.setAttribute('draggable', true);
+          } else {
+            e.removeAttribute('draggable');
+          }
+        });
+      }
+
+      /**
+       * Fired when starting to drag grid rows.
+       *
+       * @event grid-dragstart
+       * @param {Object} detail
+       * @param {Object} detail.items the items being dragged
+       */
+
+      /**
+       * Fired when the dragging of the rows ends.
+       *
+       * @event grid-dragend
+       */
+
+      /**
+       * Fired when a drop occurs on top of the grid.
+       *
+       * @event grid-drop
+       * @param {Object} detail
+       * @param {Object} detail.item tTe item of the grid row on which the drop occurred.
```

Comment:
typo: tTe

Also, inconsistent format: Here the parameters are described with full sentences ending with periods. For `grid-dragstart` it's different.

Reply:
Done.